modules/db/src/main/java/org/opencastproject/db/Activator.java

```diff
@@ -137,6 +139,36 @@ else if (pooledDataSource.getMaxIdleTime() == 0) {
     logger.info("Database connection pool established at {}", jdbcUrl);
     logger.info("Database connection pool parameters: max.size={}, min.size={}, max.idle.time={}",
       pooledDataSource.getMaxPoolSize(), pooledDataSource.getMinPoolSize(), pooledDataSource.getMaxIdleTime());
+    Statement statement = pooledDataSource.getConnection().createStatement();
+
+    long time = System.currentTimeMillis();
+    String tableName = "oc-temp" + time;
+    try {
+      runUpdate(statement, "CREATE TABLE " + tableName + " ( id BIGINT NOT NULL, test BIGINT, PRIMARY KEY (id) );");
+      runUpdate(statement, "INSERT INTO " + tableName + " VALUES (" + time + "," + time + ");");
+      runUpdate(statement, "INSERT INTO " + tableName + " VALUES (" + time + 1 + "," + time + 1 + ");");
+      runUpdate(statement, "UPDATE " + tableName + " SET test = " + time + 2 + " WHERE id = " + time + ";");
+      ResultSet rs = statement.executeQuery("SELECT FROM " + tableName + " WHERE id = " + time + 2 + ";");
+      while (rs.next()) {
+        long id = rs.getLong("id");
+        long test = rs.getLong("test");
+        if (id != time || test != time + 2) {
```

Comment:
I don't think this is actually doing what you want it to do. You are selecting the entry with `id = time + 2` but you only ever set `id = time` and `id = time + 1`. This means that you never actually go into the loop. Also, one simple insert should suffice for this test:

```suggestion
      runUpdate(statement, "INSERT INTO " + tableName + " VALUES (" + time + ", 0);");
      runUpdate(statement, "UPDATE " + tableName + " SET test = " + time + ";");
      ResultSet rs = statement.executeQuery("SELECT FROM " + tableName + ";");
      while (rs.next()) {
        long id = rs.getLong("id");
        long test = rs.getLong("test");
        if (id != time || test != time) {
```